framework/src/modules/validators/api.ts

```diff
@@ -12,20 +12,341 @@
+		apiContext: ImmutableAPIContext,
+		address: Buffer,
+	): Promise<Record<string, unknown>> {
+		const validatorsSubStore = apiContext.getStore(
+			MODULE_ID_VALIDATORS,
+			STORE_PREFIX_VALIDATORS_DATA,
+		);
+		return validatorsSubStore.getWithSchema(address, validatorAccountSchema);
+	}
+
+	public async getGenesisData(apiContext: ImmutableAPIContext): Promise<number> {
+		const genesisDataSubStore = apiContext.getStore(
+			MODULE_ID_VALIDATORS,
+			STORE_PREFIX_GENESIS_DATA,
+		);
+		const emptyKey = Buffer.alloc(0);
+		const value = await genesisDataSubStore.getWithSchema<GenesisData>(emptyKey, genesisDataSchema);
+		return value.timestamp;
 	}
 
-	public getSlotTime(_apiContext: ImmutableAPIContext, _slot: number): number {
-		return 0;
+	public async setValidatorBLSKey(
+		apiContext: APIContext,
+		validatorAddress: Buffer,
+		blsKey: Buffer,
+		proofOfPossession: Buffer,
+	): Promise<boolean> {
+		const validatorsSubStore = apiContext.getStore(
+			MODULE_ID_VALIDATORS,
+			STORE_PREFIX_VALIDATORS_DATA,
+		);
+		if (!(await validatorsSubStore.has(validatorAddress))) {
+			return false;
+		}
+
+		const blsKeysSubStore = apiContext.getStore(MODULE_ID_VALIDATORS, STORE_PREFIX_BLS_KEYS);
+		if (await blsKeysSubStore.has(blsKey)) {
+			return false;
+		}
+
+		const validatorAccount = await validatorsSubStore.getWithSchema<Record<string, unknown>>(
+			validatorAddress,
+			validatorAccountSchema,
+		);
+		if (!(validatorAccount.blsKey as Buffer).equals(INVALID_BLS_KEY)) {
+			return false;
+		}
+
+		if (!blsPopVerify(blsKey, proofOfPossession)) {
+			return false;
+		}
+
+		validatorAccount.blsKey = blsKey;
+		await validatorsSubStore.setWithSchema(
+			validatorAddress,
+			validatorAccount,
+			validatorAccountSchema,
+		);
+		await blsKeysSubStore.setWithSchema(
+			blsKey,
+			{ address: validatorAddress },
+			validatorAddressSchema,
+		);
+
+		return true;
+	}
+
+	public async setValidatorGeneratorKey(
+		apiContext: APIContext,
+		validatorAddress: Buffer,
+		generatorKey: Buffer,
+	): Promise<boolean> {
+		const validatorsSubStore = apiContext.getStore(
+			MODULE_ID_VALIDATORS,
+			STORE_PREFIX_VALIDATORS_DATA,
+		);
+		if (!(await validatorsSubStore.has(validatorAddress))) {
+			return false;
+		}
+
+		const validatorAccount = await validatorsSubStore.getWithSchema<Record<string, unknown>>(
+			validatorAddress,
+			validatorAccountSchema,
+		);
+		validatorAccount.generatorKey = generatorKey;
+		await validatorsSubStore.setWithSchema(
+			validatorAddress,
+			validatorAccount,
+			validatorAccountSchema,
+		);
+
+		return true;
+	}
+
+	public async isKeyRegistered(apiContext: ImmutableAPIContext, blsKey: Buffer): Promise<boolean> {
+		const blsKeysSubStore = apiContext.getStore(MODULE_ID_VALIDATORS, STORE_PREFIX_BLS_KEYS);
+		if (await blsKeysSubStore.has(blsKey)) {
+			return true;
+		}
+		return false;
+	}
+
+	public async getGeneratorList(apiContext: ImmutableAPIContext): Promise<{ list: Buffer[] }> {
```

Comment:
```suggestion
	public async getGeneratorList(apiContext: ImmutableAPIContext): Promise<Buffer[]> {
```
i think here no need to add object?